app/models/sign_in/user_info.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module SignIn
+  class UserInfo
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+    include ActiveModel::Serialization
+
+    ACCEPTED_GCID_TYPES = %w[
+      200ENPI 200VETS 200BRLS 200CORP 200VET360
+      200VIDM 200VLGN 200MHV 200CERNER
+    ].freeze
+
+    attribute :sub, :string
+    attribute :email, :string
+    attribute :full_name, :string
+    attribute :first_name, :string
+    attribute :last_name, :string
+    attribute :csp_type, :string
+    attribute :csp_uuid, :string
+    attribute :ial, :string
+    attribute :aal, :string
+    attribute :birth_date, :string
+    attribute :ssn, :string
+    attribute :gender, :string
+    attribute :address, :string
+    attribute :phone_number, :string
+    attribute :person_type, :string
+    attribute :icn, :string
+    attribute :sec_id, :string
+    attribute :edipi, :string
+    attribute :mhv_ien, :string
+    attribute :cerner_id, :string
+    attribute :corp_id, :string
+    attribute :birls, :string
+    attribute :gcids, :string
+    attribute :npi_id, :string
+
+    class << self
+      def from_user(user)
+        new(
+          sub: user.uuid,
+          email: user.user_verification&.user_credential_email&.credential_email || user.email,
+          npi_id: user.npi_id, full_name: full_name(user),
+          first_name: user.first_name, last_name: user.last_name,
+          csp_type: csp_type_from_mpi(user), csp_uuid: user.user_verification.credential_identifier,
+          ial: ial_level(user), aal: aal_level(user),
+          birth_date: user.birth_date, ssn: user.ssn,
+          gender: user.gender, address: user.address, phone_number: user.home_phone,
+          person_type: user.try(:person_type), icn: user.icn,
+          sec_id: user.sec_id, edipi: user.try(:edipi),
+          mhv_ien: user.try(:mhv_ien), cerner_id: user.try(:cerner_id),
+          corp_id: user.participant_id, birls: user.birls_id, gcids: validate_and_parse_gcids(user.mpi_gcids)
+        )
+      end
+
+      def ial_level(user)
+        user.user_verification.verified? ? Constants::Auth::IAL_TWO : Constants::Auth::IAL_ONE
+      end
+
+      def aal_level(user)
+        AAL::LOGIN_GOV_AAL2 if %w[idme logingov].include?(user.user_verification.credential_type)
```

Comment:
`aal` will always be 2 so we don't need to check the credential type. we can just put `aal: AAL::LOGIN_GOV_AAL2` when creating it in `from_user`